oak_runtime/src/node/http/tests.rs

```diff
@@ -20,17 +20,71 @@ use oak_abi::{
     label::Label,
     proto::oak::application::{
         node_configuration::ConfigType, ApplicationConfiguration, HttpServerConfiguration,
-        NodeConfiguration,
+        NodeConfiguration, WebAssemblyConfiguration,
     },
 };
-use std::{fs, option::Option, thread::JoinHandle};
+use std::fs;
 
 static LOCAL_CA: &str = "../examples/certs/local/ca.pem";
 static GCP_CA: &str = "../examples/certs/gcp/ca.pem";
 
+struct OakNodeForTest {}
+
+impl crate::node::Node for OakNodeForTest {
+    fn node_type(&self) -> &'static str {
+        "test"
+    }
+    fn run(
+        self: Box<Self>,
+        runtime: RuntimeProxy,
+        handle: oak_abi::Handle,
+        _notify_receiver: oneshot::Receiver<()>,
+    ) {
+        // Get invocation message that contains the response_writer handle.
+        let invocation_receiver = Receiver::<HttpInvocation>::new(ReadHandle { handle });
+
+        while let Ok(invocation) = invocation_receiver.receive(&runtime) {
+            let request = invocation.receiver.unwrap().receive(&runtime);
+            info!("received request: {:?}", request);
+
+            let resp = HttpResponse {
+                body: vec![],
+                status: http::status::StatusCode::OK.as_u16() as i32,
+                headers: Some(HeaderMap {
+                    headers: hashmap! {},
+                }),
+            };
+            invocation
+                .sender
+                .expect("Empty sender on invocation.")
+                .send(resp, &runtime)
+                .unwrap();
+        }
+    }
+}
+
+struct TestNodeFactory {
+    server_node_factory: crate::node::ServerNodeFactory,
+}
+
+impl crate::node::NodeFactory<NodeConfiguration> for TestNodeFactory {
+    fn create_node(
+        &self,
+        node_configuration: &NodeConfiguration,
+    ) -> Result<Box<dyn Node>, ConfigurationError> {
+        match &node_configuration.config_type {
+            Some(ConfigType::WasmConfig(_config)) => Ok(Box::new(OakNodeForTest {})),
```

Comment:
Am I understanding correctly that this means that whenever a wasm node is requested, this will actually create a test node, regardless of the actual configuration? And for any other node type, it just passes it through to the ServerNodeFactory? If so, it seems quite hacky, and I think we will have to change it again if we wanted to use it to test anything that does rely on Wasm nodes at all.

BTW my intention with the NodeFactory trait was actually to push it up one level and make it an explicit generic type of `Runtime` itself, so that in test we may instantiate a `Runtime<TestNodeFactory>` that may have a separate implementation just for test. Would that make things simpler for you?

Reply:
I totally agree with making `NodeFactory` an explicit generic type for `Runtime`. That would require making `RuntimeProxy` and all the pseudo-nodes and a few other things generic too. But that would allow defining a `TestNodeFactory` with a test-specific `NodeConfiguration` like the one you suggested above. So, let me try this one now... it is going to be a much bigger change.

Reply:
I agree it is a big task, though I think there a few more complications down the line, e.g. that wasm nodes will have to be compiled targeting a specific NodeConfiguration, etc., which may not be a priority for us to solve now.

For now, what about adding a method to the current NodeFactory to allow creating arbitrary nodes (by providing a function) for test? I think that is fine, because other nodes do not get direct access to the NodeFactory instance anyways.

Reply:
I think that would work, but in addition to that, we need something like `create_test_node` in the `Runtime` to call that new method in `NodeFactory`. We should be able to compile these additional methods only for test, right?

Reply:
Either that, or we need to create the NodeFactory first, and in tests return a reference to that in addition to the Runtime reference.

Anyways, a `Runtime::create_test_node` method seems fine; if we can compile it only for test then it's even better, but even if not, it's not a big deal, nothing apart from the runtime itself (or its creator) will be able to invoke it, right?

Reply:
Not sure whether `#[cfg(test)]` can be applied to methods.

Reply:
> nothing apart from the runtime itself (or its creator) will be able to invoke it, right? 
True. And it will not be exposed in the SDK. 

> Not sure whether #[cfg(test)] can be applied to methods.
I thought it is possible. I'll try.

Reply:
Another thought: perhaps it would make sense to actually expose a `Runtime::create_node` method that is invoked with the result of the NodeFactory object. In fact, I think the current code does something similar already, it just needs to be refactored slightly. No additional method for testing should be necessary.